Abnes-font: init 1.0.0

DrVoidest opened the pull request:
https://177studio.com/product/abnes-elegant-sans-serif-font/
A sci-fi font, perfect for a lock screen or other locations where a decorative font is desired.  Tested by building on local system and ensuring apps could find the font.
<!--

-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

DrVoidest committed:
Abnes-font: init 1.0.0

DrVoidest committed:
Abnes-font: init 1.0.0 - removed trailing whitespace

SigmaSquadron reviewed (commented):
Welcome to Nixpkgs! Thank you for your contribution.

I have left some comments below; mostly regarding the font's license.

You also need a [maintainer entry](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) in `maintainer-list.nix` for the `meta` attribute set to evaluate correctly. Don't forget to squash your commits after addressing all reviews.

DrVoidest committed:
Abnes-font: init 1.0.0 - added myself to the maintaners list

Abnes-font: init 1.0.0 - Moved package to correct location

Abnes-font: init 1.0.0 - Corrected upstream url and update description

DrVoidest committed:
Abnes-font: init 1.0.0 - added myself to the maintaners list

DrVoidest committed:
Abnes-font: init 1.0.0 - Moved package to correct location

DrVoidest committed:
Abnes-font: init 1.0.0 - Corrected upstream url and update description

DrVoidest committed:
Merge remote-tracking branch 'refs/remotes/origin/abnes-font' into abnes-font

DrVoidest committed:
Abnes-font: init 1.0.0 - Moved pkg to the right location for real this time

DrVoidest committed:
Abnes-font: init 1.0.0 - Now named the correct thing

DrVoidest commented:
I apologize for such a messy I made for tree you all. I tried to squash the commits, but it looks like I just worsened it. 177 Studio allows redistribution for personal use from the following sites: https://befonts.com/author/177studio, https://www.fontspace.com/177studio, https://www.dafont.com/kenz.d8820 and I as such changed the fetch url to befonts. I have also contacted them for explicated permission to redistribute the font on nixos.

SigmaSquadron commented:
If you've contacted them, then we should wait for their reply. I personally doubt you'll get permission though.

DrVoidest commented:
Hey, maybe we will get lucky. Thank you for spending your spare time helping me with this PR, everyone.
